spec/lib/payment_processor/go_cardless/error_processing_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'rails_helper'
+
+module PaymentProcessor
+  module GoCardless
+    describe ErrorProcessing do
+
+      let(:client) do
+        GoCardlessPro::Client.new(
+          access_token: Settings.gocardless.token,
+          environment:  Settings.gocardless.environment.to_sym
+        ) 
+      end
+
+      before :each do
+        allow(Rails.logger).to receive(:error)
+      end
+
+      subject{ described_class.new(@error).process }
+
+      describe 'with a GoCardless internal error', :pending do
+
+        before :each do
+          begin
+            client.redirect_flows.create(params: {})
+          rescue GoCardlessPro::GoCardlessError => e
+            @error = e
+          end
```

Comment:
Not sure I quite understand. If this is a unit test then what is raised isn't a concern. You'd only want to see how the class responds to different error types by instantiating the class with different test doubles. Am I missing something?